### PR TITLE
chore(ts-migration): finishes migration of configure and configure re…

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request adjust previous already migrated to TS:

- configure ( migrates stories, adds named function  )
- configure related items ( adds ConfigureRelatedItemsWidget exports, adds named function )

### CODE REVIEW

File: src/widgets/configure/configure.ts [hunk 1]
```diff
@@ -11,12 +11,12 @@ export type ConfigureWidgetOptions = PlainSearchParameters;
...
 export type ConfigureWidget = WidgetFactory<{}, ConfigureWidgetOptions>;
 
-const configure: ConfigureWidget = widgetParams => {
+const configure: ConfigureWidget = function configure(widgetOptions) {
   // This is a renderless widget that falls back to the connector's
   // noop render and unmount functions.
   const makeWidget = connectConfigure(noop);
 
-  return makeWidget({ searchParameters: widgetParams });
+  return makeWidget({ searchParameters: widgetOptions });
```
Comment (eunjae-lee) [inline src/widgets/configure/configure.ts:19]: Is this because of ConfigureWidget _Options_ ?

Comment (nunomaduro) [reply src/widgets/configure/configure.ts:19]: Actually, we have `widgetParams` in some widgets, and `widgetOptions` in others. I am just to unify the style.

Comment (eunjae-lee) [reply src/widgets/configure/configure.ts:19]: So is [this](https://github.com/algolia/instantsearch.js/blob/chore/typescript-migration-configures/src/types/widget.ts#L202:L202) going to be changed to `widgetOptions` in the end? (meaning the public signature changes along with the docs)

Comment (nunomaduro) [reply src/widgets/configure/configure.ts:19]: Nice catch. It's the other way around. It should be widgetParams. Going to update the pull request.

Comment (nunomaduro) [reply src/widgets/configure/configure.ts]: Done, updating the named `widgetParams` vs `widgetOptions` in other pull request.

Comment (eunjae-lee) [reply src/widgets/configure/configure.ts]: Just confirmed 👍

**Comment by nunomaduro** [comment]
Waiting for https://github.com/algolia/instantsearch.js/pull/4388 to be merged first.

**Pull request merged**